src/main/java/com/google/sps/servlets/NewClass.java

```diff
@@ -0,0 +1,38 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceConfig;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that creates a new class Datastore. */
+@WebServlet("/newclass")
+public final class NewClass extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    // navigate to /_ah/admin to view Datastore
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    System.setProperty(
+        DatastoreServiceConfig.DATASTORE_EMPTY_LIST_SUPPORT, Boolean.TRUE.toString());
+
+    String className = request.getParameter("className");
+    List<String> emptyQueue = Collections.emptyList();
+
+    Entity classEntity = new Entity("Class");
+    classEntity.setProperty("owner", "");
+    classEntity.setProperty("name", className);
+    classEntity.setProperty("beingHelped", "");
+    classEntity.setProperty("studentQueue", emptyQueue);
+
+    datastore.put(classEntity);
```

Comment:
After the class is created would this servlet have to return the class key to the user?

Reply:
I was thinking the class key could be given to the user somewhere on the dashboard (where it should redirect after they click to create the new class) instead of being returned by the servlet, but we should probably discuss this more as we put everything together.

Reply:
Okay cool! Just wanted to double check.